### PR TITLE
Create option for unison exe that will write RTS stats to a file

### LINKED ISSUE #3597: Structured benchmarking/performance output from transcript runs
* There should be some way to run a transcript on an existing codebase and acquire any relevant statistics about the run, e.g. memory usage, allocations, run-time, cpu usage, etc.
* The data must be output in a structured format that's easy to consume/query (e.g. JSON, csv)

### PR DESCRIPTION
**Pull request by tstat**
fixes #3597

### CODE REVIEW

File: unison-cli/package.yaml [hunk 1]
```diff
@@ -117,7 +117,7 @@ executables:
         other-modules: Paths_unison_cli
     source-dirs: unison
     main: Main.hs
-    ghc-options: -threaded -rtsopts -with-rtsopts=-I0 -optP-Wno-nonportable-include-path
+    ghc-options: -threaded -rtsopts "-with-rtsopts=-I0 -T" -optP-Wno-nonportable-include-path
```
Comment (ChrisPenner) [inline unison-cli/package.yaml]: I don't have any information on it, but I'm guessing that collecting and maintaining run time stats isn't completely free, does it make sense to enable this for our production executables? Should we instead find a way just to enable it for our transcript executable? Or maybe using a build flag?

**Review by ChrisPenner — APPROVED** [review]
Looks good, just had a question about enabling this on our main executable vs just a transcripts one

File: unison-cli/package.yaml (hunk 1 above)
Comment (mitchellwrosen) [reply unison-cli/package.yaml]: It's very cheap & normal to have on in production for monitoring stats

**Review by aryairani — APPROVED** [review]
Looks good to go after resolving conflicts.

**Pull request merged**